ls1intum/Artemis
# `Communication`: Automatically delete associated channel on deletion

### Checklist
#### General


#### Server

### Motivation and Context


Channels that are associated with a lecture, exercise or exam can only be manually deleted now, but are not automatically deleted when the according lecture/exercise/exam is deleted.

### Description

If a channel is associated with a lecture/exercise/exam, then `ConversationService.deleteConversation()` is used to delete the associated channel if the lecture/exercise/exam gets deleted.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with messaging enabled

1. Find or create a lecture with a channel
2. Delete the lecture
3. Confirm, that the channel was deleted as well (navigate to the messages tab and browse channels)
4. Repeat 1-3 for each exercise type and exams

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage






#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ExerciseDeletionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5099-5/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/ExerciseDeletionService.html) | 96% | ✅ |
| [LectureService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5099-5/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/LectureService.html) | 100% | ✅ |
| [ExamDeletionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5099-5/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.exam/ExamDeletionService.html) | 100% | ✅ |
| [ChannelService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5099-5/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis.conversation/ChannelService.html) | 90% | ✅ |
